### DISCUSSION TITLE
How to better document deprecated met/rxn ids

### DISCUSSION THREAD

**Discussion by haowang-bioinfo** [SysBioChalmers/Human-GEM#615] [Ideas]
The current workflow is becoming inadequate with the increasing curation amount, and  an upgraded workflow with "merge queue" feature might be needed. 

Ideas and suggestions are welcome.

**Comment by feiranl** [comment]
Should we introduce ChatGPT to help us to have a first check? I am not sure wether this will work, but basically just like the chatbot currently used in other websites. If the issue is well described and the ChatGPT can help us to first check wether the issue and proposed solution is ok? but then this required the person who raises the issue and solution very well. Then as for the merge of PR, we need to have more tasks such as organ-specific tasks. But I am not sure wether the server load can handle. @mihai-sysbio

**Reply by mihai-sysbio** [reply]
> Should we introduce ChatGPT to help us to have a first check? I am not sure wether this will work, but basically just like the chatbot currently used in other websites. If the issue is well described and the ChatGPT can help us to first check wether the issue and proposed solution is ok? 

I've done some investigation in https://github.com/SysBioChalmers/Human-GEM/issues/483 . In summary, so far ChatGTP tends to imagine too much for it to be considered accurate. So far I haven't managed to phrase questions so as to obtain concrete, reference-backed answers, that would be better than an API request from a specialised service.

**Reply by haowang-bioinfo** [reply]
sounds good, any specific plan do you have? @feiranl

**Comment by feiranl** [comment]
Besides this, should we also give one report for every major release? such as summaries all changes and reasons in one txt file. I am recently working on the organ-specific tasks, but found many reactions are deprecated, but the reason for them are dispersed in the many issues pages. So I want to propose a report file for each major release. So if we combine all reports, then we can clearly see all model changes.
I can give one example:

`ELAIDCRNtr`(rxn in Recon3D model) was identified as a metabolic essential Task, however, this one matched Human-GEM rxn `MAR11269` which was removed in #344,  since it is a duplicate with `MAR10178`. Without manual check, it is difficult for me to identify the `ELAIDCRNtr` should mapped to `MAR10178`.
@Devlin-Moyer the recent PR also removed many duplicates, so it is great to merge all annotations and cross references.

**Reply by mihai-sysbio** [reply]
While possible, I think this kind of report will be hard to create. I have not seen anything like it - to be able to provide a comprehensive report of every small change in a model. Such a report also runs the risk of being so long for some of the releases.

Regarding the particular example or RECON3D and Human-GEM mapping, my approach would be to look back into [`reactions.tsv` in release `1.7`](https://github.com/SysBioChalmers/Human-GEM/blob/02613bf66175f74fe29a6fdc5d382d9c0580f4a2/model/reactions.tsv#LL11161C1-L11161C73). This release contains the mapping between models, and is also old enough that it is way prior to many of the manual curations.

**Reply by haowang-bioinfo** [reply]
@feiranl as you pointed out, so far the reasoning logic of model changes is dispersed in issues and PR messages.

So far `reactions.tsv` and `metabolites.tsv` are the log files which basically preserve the change outcome every time. Here probably you are suggesting a better way of documenting the changes, by integrating the logic into these log files? please correct me if I interpret wrongly

**Reply by haowang-bioinfo** [reply]
and this topic might deserve a dedicated discussion, which doesn't appear to be related to "merge queue"

**Reply by mihai-sysbio** [reply]
Hmm, one could add a column with the issue numbers where that line was last edited. This part can indeed be scripted with a GH action.

**Reply by haowang-bioinfo** [reply]
adding a new column to the two tsv files indeed addresses the issue at some degree. The content of this new column probably worths further consideration.

**Reply by feiranl** [reply]
Agree! and Now we are adding the retired Human-GEM IDs due to duplicates to the retireID, somehow to document this a little bit.

**Reply by mihai-sysbio** [reply]
When thinking about that column, please keep in mind:
- a row might receive curation multiple times, and it ought to be possible to distinguish these
- the file is a TSV and should still be relatively lightweight
- the column should be machine-readable (e.g., imagine this information could end up on Metabolic Atlas in a _Curation history_ section)

**Reply by haowang-bioinfo** [reply]
> the column should be machine-readable (e.g., imagine this information could end up on Metabolic Atlas in a Curation history section)

it would be a huge bonus if this column could end up in a `Curation history` section on Metabolic Atlas

**Reply by feiranl** [reply]
Totally agree. It would be great to say where and when the rxn/met was added, when and how it was modified, which issue and PR is related to that and the reference.

**Reply by haowang-bioinfo** [reply]
in fact, by a quick looking, it's found that most of the pending PRs are resolving duplicate mets and rxns, a scenario highly relevant to the topic here

actually, across the accumulated 300+ PRs, many are also resolving duplicate mets and rxns.

**Reply by haowang-bioinfo** [reply]
> we are adding the retired Human-GEM IDs due to duplicates to the retireID, somehow to document this a little bit.

yes, this is straightforward solution, i.e. a natural selection. Because  `rxnRetired`, `metRetired` for mets, is an existing column, one can just append duplicated ids to the end and delimted by `;`. This probably can be implemented right away in the pending PRs? @feiranl @mihai-sysbio @Devlin-Moyer

**Reply by haowang-bioinfo** [reply]
> one could add a column with the issue numbers where that line was last edited. This part can indeed be scripted with a GH action.

this could be a nice feature.

**Reply by Devlin-Moyer** [reply]
Oh yea that does make sense; I never touched that column because I wasn't 100% sure what it was for (my guess was that at some point you changed the general format of IDs and reassigned a bunch of metabolite/reaction IDs without changing anything about the metabolites or reactions themselves) but I can definitely start doing that
So far, I'd been trying to put the "removed _ for being a duplicate of _" in the commit or PR messages, but I agree that the retired columns would be a much nicer way to systematically record that

**Reply by haowang-bioinfo** [reply]
a tentative implementation was made in #666 as an example, any feedback is welcome

**Comment by mihai-sysbio** [comment]
Another slightly different idea when doing curation is to have a script that fetches all the commit hashes and timestamps for each reaction/metabolite in the model.

**Reply by haowang-bioinfo** [reply]
sounds good - go ahead please

**Reply by edkerk** [reply]
As a sort of `blame`?

**Reply by mihai-sysbio** [reply]
Yes. It can take a bit to blame Human-GEM.yml (314k lines), but anyways sifting through that output is really not that useful, since some edits can affect only formatting and not content.
